### PR TITLE
Create portal login

### PR DESCRIPTION
**Pull request by MohammedIrheem**
**Create a portal login page**
  - This page will contain an input field and button for submitting.
  - I used ant Design  Form 
  - I used Axios  to send a request to the server, and check the response status:
      - if the status 301 then redirect to the portal page.
      - if the status 400 or 401 will display an error message

### CODE REVIEW

File: client/src/components/container/portal/LogInPage/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      <div className='portal-contant'>
+        <h1 className='title'>Welcome to <span>GSG Events portal login page</span></h1>
+        <Form
+          className='main-form'
+          onFinish={this.onFinish}
+        >
+          <Form.Item
+            className='input-field'
+            name="pinCode"
+            rules={[{ message: 'Please input your pin-code!' }]}
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:55]: **add required to rules :** 
```suggestion
           rules={[{ required: true, message: 'Please input your pin-code!'}]}>
```

File: client/src/components/container/portal/LogInPage/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import { Form, Input, Button, Spin, Alert, message } from 'antd';
+import axios from 'axios';
+
+import './style.css'
+export class PortalLogin extends Component {
+  state = {
+    isLoade: false,
+    picCode: '',
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:9]:
```suggestion
    pinCode: '',
```

File: client/src/components/container/portal/LogInPage/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    msg: '',
+    error: false,
+  }
+
+  handleChange = ({target:{value}}) => {
+    this.setState({ pinCode: value, error: false });
+  };
+
+  onFinish = ({ pinCode }) => {
+    axios.post('/api/v1/portal/login', { pinCode}).then(({ data }) => {
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:20]: set `isLoaded:true : true` , before axios request.

File: client/src/components/container/portal/LogInPage/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          <Form.Item
+            className='input-field'
+            name="pinCode"
+            rules={[{ message: 'Please input your pin-code!' }]}
+          >
+            <Input.Password placeholder='Enter your pin code' onChange={this.handleChange}/>
+          </Form.Item>
+          <Form.Item >
+            <Button type="primary" htmlType="submit">
+              Login
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:61]:
```suggestion
               {isLoade ? (<Spin />) : Login}
```

File: client/src/components/container/portal/LogInPage/index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+          msg: data.msg,
+          pinCode: '',
+        })
+      } else if (data.status === 401) {
+        this.setState({
+          error: true,
+          msg: data.msg,
+          pinCode: '',
+        })
+      }
```
Comment (AlaaTaima) [inline client/src/components/container/portal/LogInPage/index.js:37]: all cases of error will be handled at the same way here so you don't need to check for every case.
```suggestion
      } else {
        this.setState({
          error: true,
          msg: data.msg,
          pinCode: '',
        })
      }
```

File: server/controller/routes/portal/portalLogainPage.js [hunk 6]
```diff
@@ -13,7 +12,7 @@ const pinCodeValidation = (req, res, next) => {
       next();
     })
     .catch((err) => {
-      res.json({ msg: err.message });
+      res.json({ msg: 'Invalid Inputs!', status: 400 });
```
Comment (AhmedSafi97) [inline server/controller/routes/portal/portalLogainPage.js:15]:
```suggestion
      res.status(400).json({ msg: err.message });
```

File: server/middleware/checkPinCode.js [hunk 7]
```diff
@@ -6,14 +6,19 @@ const checkPinCode = (req, res, next) => {
...
       err.status = 401;
-      next(err);
+      throw err;
     } else {
       next();
     }
-  }).catch(next);
+  }).catch((err) => {
+    if (err.status === 401) {
+      res.json(err);
```
Comment (AhmedSafi97) [inline server/middleware/checkPinCode.js:17]:
```suggestion
      res.status(401).json(err);
```

File: client/src/components/container/portal/LogInPage/index.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        this.setState({
+          error: true,
+          msg: data.msg,
+          pinCode: '',
+        })}
+      this.setState({ isLoade: false })
+    }).catch(() => {
+      this.setState({ serverError: "Internal server error !!", isLoade: false })
+    })
+  };
```
Comment (AhmedSafi97) [inline client/src/components/container/portal/LogInPage/index.js:35]: **you should handle the errors in the catch block, [see axios doc ](https://github.com/axios/axios#handling-errors)**
```suggestion
onFinish = ({ pinCode }) => {
    this.setState({ isLoade: true });
    axios
      .post("/api/v1/portal/login", { pinCode })
      .then(({ data }) => {
        this.props.history.push("/portal/attendance");
        message.success(data.msg, 10);
      })
      .catch(
        ({
          response: {
            data: { msg },
          },
        }) => {
          this.setState({
            error: true,
            msg,
            pinCode: "",
            isLoade: false,
          });
        }
      );
  };
```

File: server/controller/routes/portal/portalLogainPage.js [hunk 9]
```diff
@@ -29,7 +28,7 @@ const getHashedPinCode = (req, res, next) => {
 const createPortalToken = (req, res, next) => {
   const token = jwt.sign({ id: req.adminId }, process.env.SECRET_KEY_PORTAL);
   res.cookie('portalToken', token);
-  res.json({ msg: 'Portal, logged in successfully' });
+  res.json({ msg: 'Portal, logged in successfully', status: 301 });
```
Comment (AhmedSafi97) [inline server/controller/routes/portal/portalLogainPage.js:31]: if you stick to what I mentioned above then you won't need the 301 status 

```suggestion
  res.json({ msg: 'logged in successfully' });
```

**Pull request merged**